eng/common/scripts/Scan-files-against-links.ps1

```diff
@@ -0,0 +1,43 @@
+param (
+    # The deleted files from the PR.
+    [string[]] $deletedFiles,
+    # The renamed files from the PR.
+    [string[]] $renamedFiles,
+    # The markdown files from the PR.
+    [string[]] $changedMarkdowns
+)
+
+$beforeRenameFiles = @()
+# Retrieve the renamed from files.
+foreach($file in $renamedFiles) {
+    if ($file -match "^rename from (.*)$") {
+        $beforeRenameFiles += $file -replace "^rename from (.*)$", '$1'
+    }
+}
+# A combined list of deleted and renamed files.
+$relativePathLinks += ($deletedFiles + $beforeRenameFiles)
+
+# Scan all markdowns and find if it contains the deleted or renamed files.
+$markdownContainLinks = @()
+foreach ($f in (Get-ChildItem -Path ./ -Recurse -Include *.md)) {
+    foreach($l in $fullPathLinks) {
+        $content = Get-Content -Path $f -Raw
```

Comment:
Lets move this out of the inner loop as we don't need to read the file content for ever link we are checking. We only need to that once per file.